Payloads - add crop spray and pull out generic actuators

hamishwillee opened the pull request:
This adds a section for agricultural drones/crop sprayers to the payloads doc.

As part of this I pulled the generic actuator control into its own section and added a subsection on how you use this setup in QGC for mission (i.e. with great difficulty because there is no mission item).

FYI @akshata-01 

@junwoo091400 
- FMI, any progress on supporting winches in payload delivery? I'd say they are just as common as grippers.
- If you're looking for a useful job to do, would it be possible to add a mission item to QGC for set_actuator? There is a set_servo one there, but that isn't as generic (I'm not even sure whether/how it works for PX4). Right now it is very hard to use generic actuator control in missions except through MAVSDK.

hamishwillee commented:
PS, I hope to merge this tomorrow. It is much better structurally but I'm hoping someone else can sanity check it as is hard for me to be objective just after writing something

julianoes commented on en/payloads/README.md:
```diff
@@ -19,50 +19,80 @@ The MAVSDK provides simple APIs to use this protocol for both [standalone camera
+- The [Generic Actuator Control with MAVLink Command](#generic-actuator-control-with-mavlink-command) section explains how you can connect flight controller outputs to your sprayer and trigger them using the `MAV_CMD_DO_SET_ACTUATOR`.
+- You can define the area to spray using a [Survey pattern](https://docs.qgroundcontrol.com/master/en/PlanView/pattern_survey.html), or you can define the grid to fly using waypoints.
+  In either case, it is important to ensure that the vehicle flight path provides adqequate coverage for your payload.
+- You should add a `MAV_CMD_DO_SET_ACTUATOR` to your mission before and after the survey pattern in order to enable and disable the sprayer.
```
Does this command actually work? Would be good thing to verify it with a MAVSDK example...

hamishwillee replied:
@julianoes I'm going to assume that they do. Firstly because @junwoo091400 says they should, secondly because there is the MAVSDK example below that I understand was used to test this.

hamishwillee replied:
And lastly, because the code to support this is present. It should have been tested when that went in. This section is already present in docs - I've just moved it around. So retesting would be quite a bit of work for me, and we're no worse off if I do not.

julianoes commented on en/payloads/README.md:
```diff
@@ -79,6 +109,28 @@ The outputs that are to be controlled are specified in the [Actuators](../config
+
+1. Create a mission that uses a placeholder mission item where you want the actuator command (such as [MAV_CMD_DO_SET_SERVO](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_SERVO)).
+2. Export the plan to a file and open the file
+3. Find the placeholder mission item(s) using the command number; for `SET_SERVO` this will look like `"command": 183,`.
+4. Update the fields in the command to match the desired [MAV_CMD_DO_SET_ACTUATOR](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_ACTUATOR) command values (for example, the command number will change to `187`)
```
Again, do these work? Would be nice to test.

junwoo091400 commented:
Btw SET_ACTUATOR should work in PX4 :thinking: (I had a related follow-up PR, but it's not critical: https://github.com/PX4/PX4-Autopilot/pull/20392)

junwoo091400 commented:
> * would it be possible to add a mission item to QGC for set_actuator?

![image](https://user-images.githubusercontent.com/23277211/210653994-858182d3-af08-45fd-afca-e66ee747469d.png)

I just tried with QGC latest App Image, and you can select 'Set actuator' waypoint using the 'Advanced' category, just like it is done in Gripper Mechanism:

![image](https://user-images.githubusercontent.com/23277211/210654104-39f408bf-761d-4a3b-8077-53d3ae54e105.png)

Maybe you missed this? @hamishwillee

hamishwillee commented:
> Maybe you missed this? @hamishwillee

Thanks @junwoo091400 No, it just isn't in the release build which I did test. Yes is present in the daily.

FWIW The item supports 4 actuators. PX4 v1.13 supports 3 actuators. PX4 main supports 6 in a single message, unlimited by using the index. IMO this should be updated to allow 6 actuators to be set.

hamishwillee commented:
Thanks all for your reviews. I am trusting that this actually works (as per comments). I've updated this to explain how to set the "Set actuator" mission item by modifying a waypoint, and noted that the user should select a more recent QGC if it is not present.

Also added a bit more explanation of why generic actuator control is not quite as good.

Pretty happy with this. Happy to take fixes as post process.
